lib/rules/disallow-spaces-in-function-declaration.js

```diff
@@ -75,6 +75,9 @@ module.exports.prototype = {
         var beforeOpeningCurlyBrace = this._beforeOpeningCurlyBrace;
 
         file.iterateNodesByType(['FunctionDeclaration'], function(node) {
+            if (!node.id) {
```

Comment:
Please move the description of this PR to an issue (the explanation of why this needs to happen) and then reference the issue number in a comment above this line. The reasoning as to why we need to do this should be captured in the code.